server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java

```diff
@@ -321,6 +324,21 @@ public void run() {
     LOG.info("Shutting down");
   }
 
+  private Map<String,List<HostAndPort>> getIdleCompactors() {
+    Map<String,List<HostAndPort>> queuesAndCompactors =
+        ExternalCompactionUtil.getCompactorAddrs(getContext());
+    RUNNING_CACHE.values().forEach(rc -> {
+      List<HostAndPort> compactors = queuesAndCompactors.get(rc.getQueueName());
+      if (compactors != null
+          && compactors.remove(HostAndPort.fromString(rc.getCompactorAddress()))) {
```

Comment:
Repeated calls to remove would be cheaper w/ a Set instead of a List.  If possible it would be nice to change it to a set if doing so does not negatively impact other code.

Reply:
Implemented in 6cf5f76